Reqnroll/InternalPlugins/DryRunBindingInvokerPlugin.cs

```diff
@@ -0,0 +1,29 @@
+﻿using Reqnroll.Bindings;
+using Reqnroll.CommonModels;
+using Reqnroll.EnvironmentAccess;
+using Reqnroll.Plugins;
+using Reqnroll.UnitTestProvider;
+
+namespace Reqnroll.InternalPlugins;
+
+internal class DryRunBindingInvokerPlugin : IRuntimePlugin
+{
+    internal const string DryRunEnvVarName = "REQNROLL__DRY_RUN";
+
+    public void Initialize(RuntimePluginEvents runtimePluginEvents, RuntimePluginParameters runtimePluginParameters, UnitTestProviderConfiguration unitTestProviderConfiguration)
+    {
+        runtimePluginEvents.CustomizeGlobalDependencies += (_, args) =>
+        {
+            var environment = args.ObjectContainer.Resolve<IEnvironmentWrapper>();
```

Comment:
This line causes me to pause. Ideally we do not resolve any interfaces from within a `CustomizeGlobalDependencies` method. Doing so would prevent any other plug-in from substituting it's own `IEnvironmentProvider`. 

I'm not sure what to suggest as an alternative. Waiting any later in the execution sequence and we lose the opportunity to substitute the DryRunInvoker.

I would like to see a discussion with the core group. It may be an entirely acceptable risk. Or perhaps we push for a direction in which things like configuration and environment variables are resolved/finalized before plugins customize dependencies.

Reply:
Great point! I _could_ just go straight for resolving the env var directly here... (And figure out how to adjust the unit tests accordingly.) But something about doing that doesn't seem quite right either if this IEnvironmentProvider abstraction exists 🤔

Reply:
A new approach has been pushed to this PR.